homeassistant/components/emoncms/sensor.py

```diff
@@ -146,20 +159,9 @@ def __init__(
         elem = {}
         if self.coordinator.data:
             elem = self.coordinator.data[self.idx]
-        if name is None:
-            # Suppress ID in sensor name if it's 1, since most people won't
-            # have more than one EmonCMS source and it's redundant to show the
-            # ID if there's only one.
-            id_for_name = "" if str(sensorid) == "1" else sensorid
-            # Use the feed name assigned in EmonCMS or fall back to the feed ID
-            feed_name = elem.get("name", f"Feed {elem.get('id')}")
-            self._attr_name = f"EmonCMS{id_for_name} {feed_name}"
-        else:
-            self._attr_name = name
-        self._value_template = value_template
+        self._attr_name = f"{name} {elem[FEED_NAME]}"
         self._attr_native_unit_of_measurement = unit_of_measurement
-        self._sensorid = sensorid
-
+        self._attr_unique_id = f"{entry_id}-{elem[FEED_ID]}"
```

Comment:
Double check: previous sensors didnt have unique ids?

Reply:
Yea I can remember unique_id and emoncms PR existed, but not sure what happened with it. Thanks